app/models/ability.rb

```diff
@@ -0,0 +1,10 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :manage, :all if user.role == 'admin'
+
+    can :destroy, Post, author: user
+    can :destroy, Comment, author: user
```

Comment:
- [ ] [OPTIONAL] The owner of a post can perform all view, update, or destroy their own post if Post.author == user, kindly consider changing `destroy` to `manage` for this to work as expected